pkg/webui/console/containers/gateway-onboarding-form/gateway-provisioning-form/gateway-claim-form-section/index.js

```diff
@@ -0,0 +1,150 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React, { useEffect } from 'react'
+import { defineMessages } from 'react-intl'
+import { useFormikContext } from 'formik'
+
+import frequencyPlans from '@console/constants/frequency-plans'
+
+import Form from '@ttn-lw/components/form'
+import Input from '@ttn-lw/components/input'
+import Notification from '@ttn-lw/components/notification'
+import SubmitBar from '@ttn-lw/components/submit-bar'
+import FormSubmit from '@ttn-lw/components/form/submit'
+import SubmitButton from '@ttn-lw/components/submit-button'
+import Link from '@ttn-lw/components/link'
+
+import { GsFrequencyPlansSelect as FrequencyPlansSelect } from '@console/containers/freq-plans-select'
+
+import { selectGsConfig } from '@ttn-lw/lib/selectors/env'
+import sharedMessages from '@ttn-lw/lib/shared-messages'
+import tooltipIds from '@ttn-lw/lib/constants/tooltip-ids'
+import getHostFromUrl from '@ttn-lw/lib/host-from-url'
+
+const { enabled: gsEnabled, base_url: gsBaseURL } = selectGsConfig()
+
+const m = defineMessages({
+  claimWarning:
+    'We detected that your gateway is a <strong>Managed Gateway</strong>. To claim this gateway, please use the claim authentication code printed on the <link1>inside of the mounting lid</link1> or <link2>scan the QR code</link2> to claim instantly.',
+  preselectCommonOption:
+    'We preselect a common option for the region you are in. Please verify the gateway operates on the selected frequency plan.',
+})
+
+const getCupsUri = gsUrl => {
+  try {
+    const gtwUrl = new URL(gsUrl)
+
+    return `${gtwUrl.protocol}//${gtwUrl.hostname}:443`
+  } catch (err) {
+    // Ignore any errors related to the URL composition.
+  }
+
+  return ''
+}
+
+const initialValues = {
+  authenticated_identifiers: {
+    authentication_code: '',
+    gateway_eui: '',
+  },
+  target_gateway_id: '',
+  target_frequency_plan_id: '',
+  target_gateway_server_address: gsEnabled ? getHostFromUrl(gsBaseURL) : '',
+  cups_redirection: {
+    target_cups_uri: gsEnabled ? getCupsUri(gsBaseURL) : '',
+  },
+}
+
+// This is the TrackNet prefix that all TTIGs use.
+// const TRACKNET_PREFIX = '58A0CBFFFE'
+
+const GatewayClaimFormSection = () => {
+  const { values, addToFieldRegistry, removeFromFieldRegistry } = useFormikContext()
+
+  const isTTIG = values?._inputMethod === 'tts'
+
+  // Register hidden fields so they don't get cleaned.
+  useEffect(() => {
+    const hiddenFields = ['target_gateway_server_address', 'cups_redirection']
+    addToFieldRegistry(...hiddenFields)
+    return () => removeFromFieldRegistry(...hiddenFields)
+  }, [addToFieldRegistry, removeFromFieldRegistry])
```

Comment:
In the submit function you have access to the cleaned values:
```
const handleSubmit = useCallback(
    async (values, { }, cleanedValues) => { },
    []
)
```

So, I think this useEffect() shoudn't be necessary.

Reply:
I copy pasted most of the registration/claiming code from ENT repo and looks like this code is that these fields are added to the `fieldRegistry` so that they will be present in the `cleanedValues` in the `handleSubmit` function. Once I remove the `useEffect`, `target_gateway_server_address ` and `cups_redirection` are not present in the clean values. This is the submit function that his form is using:

```
const handleSubmit = useCallback(
    ({ _inputMethod, ...values }, _, cleanValues) => {
      if (_inputMethod !== 'register') {
        return handleClaimSubmit(values, cleanValues)
      }
      return handleRegistrationSubmit(values, cleanValues)
    },
    [handleClaimSubmit, handleRegistrationSubmit],
  )

  const downloadLns = useCallback(() => {
    if (lnsKey) {
      const content = composeDataUri(`Authorization: Bearer ${lnsKey}\r\n`)
      downloadDataUriAsFile(content, 'tc.key')
    }
  }, [lnsKey])
```